InClass/Callbacks/exercise-1/exercise.js

```diff
@@ -11,3 +11,17 @@ Update your code to make the colour change every 5 seconds to something differen
 Prefer to work on a codepen? https://codepen.io/makanti/pen/abOreLg
 ================
 */
+let body = document.querySelector("body");
+let colours = ["green", "pink", "purple", "orange"];
+let i = 0;
+
+// Changes the background to goldenrod after 2 seconds of loading the page.
+setTimeout(() => {
+  body.style.backgroundColor = "goldenrod";
+}, 2000);
+
+function changeBgColour() {
+  body.style.backgroundColor = colours[i];
```

Comment:
Hi
my suggestion here is not using global variable `i`. 
in general, we should try to avoid using globals as possible, it's not really safe and such code prone for unexpected buggy surprises. Also, what the line below (line #25) really doing is just incrementing `i` every next time.

So, here we can generate that index position randomly and that would probably be the best option:

```js
let i = Math.floor(Math.random() * colours.length);
body.style.backgroundColor = colours[i];

/*
Math.random() gives a random value from 0 to 0.999... 
colours.length in out current situation is 4
so max(i) we can get is 0.999 * 4 = 3.996
Floor(3.996) will give us us 3
Hence `i` in this case can only be in range between 0 and 3 which suits our need.
And yet, it's local variable in scope of function changeBgColour() only.
*/

```